pkg/ioutils/readers.go

```diff
@@ -189,6 +190,9 @@ func (r *bufReader) drain() {
 		reuseCount++
 		r.wait.Signal()
 		r.Unlock()
+		//allow or force Go scheduler to switch context, without explicitly
+		//forcing this will make it hang when using gccgo implementation
+		runtime.Gosched()
```

Comment:
Not sure if we want to introduce build flags, this can make code complex to read. If I understand, you want condition compilation like C #ifdef abc ... #endif?
I can do TODO so that we can scan code from time to time to get rid of TODO as possible.